### ISSUE TITLE
Add Documenter to NNlib

### ISSUE THREAD

**Issue by mprat** [FluxML/NNlib.jl#137]
It would be awesome for NNlib to have auto-generated documentation like many other wonderful Julia packages (including Flux.jl). I know you can always look at source, but documentation is really nice sometimes!

I would love to take a stab at integrating Documenter to NNlib as part of Hacktoberfest, if folks are willing to be patient with me and help out when I get stuck! I'll start working on a local branch for this and see how it goes.

Do most Julia packages host documentation on Github Pages?

**Comment by MikeInnes** [comment]
Agreed! Sorry we missed hacktoberfest, but if you're still willing to give this a go we'd be happy to help.

**Comment by CarloLucibello** [comment]
we have docs now
